### DISCUSSION TITLE
Consider signing of tags and commits

### DISCUSSION THREAD

**Discussion by inglor** [dundee/gdu#110] [Ideas]
Hi @dundee 

Since you already have PGP uploaded to github [1] please consider signing your tags (releases) at least, if not all your commits. The benefits for this are detailed here [2]

My plan is to package this on Arch Linux repos and it would assist on making sure the tagged source is valid.

[1]: https://github.com/dundee.gpg
[2]: https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification

**Comment by dundee** [comment]
Hi @inglor,
all tags should be already signed for some time.

Are you planning to package it for community repo? There are already packages in AUR.

**Comment by inglor** [comment]
Yes - sorry it might not be clear since I use `inglor` in github as username.

I'm `artafinde` in AUR and Arch Linux https://wiki.archlinux.org/title/Trusted_Users#Active_Trusted_Users

As for the commit tag signed see an example from `gitea` [1]
![image](https://user-images.githubusercontent.com/301857/144584075-a379f132-8649-4d9c-af02-05a367940c85.png)

That way Arch Linux tools can verify the commit of the tag directly during build.

Some more info from https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work

[1]: https://github.com/go-gitea/gitea/releases

**Comment by danie-dejager** [comment]
I see only the tag is signed
![image](https://user-images.githubusercontent.com/13314239/144586999-c184d829-3403-49a8-8fac-b51f0f7499e4.png)
![image](https://user-images.githubusercontent.com/13314239/144587109-b5a214a3-d1c6-4a9a-afdf-00799b188e16.png)

**Comment by dundee** [comment]
@inglor Ok, I will try to sign all commits from now.

Great! Looking forward to seeing gdu in community.

**Reply by dundee** [reply]
I was myself considering becoming a trusted user but I don't maintain many [packages](https://aur.archlinux.org/packages/?SeB=m&K=dundee)

**Comment by inglor** [comment]
@dundee new release with signed commits looks awesome! Package updated and it's ready for repos, just waiting for the keyring to be updated with my key, takes a bit of time.

Thanks for sorting this out this quickly!
